docs/resources/samples/write-large-dataset.md

```diff
@@ -141,6 +143,119 @@ function getRandomString(length: number): string {
+
+1. Create a workbook in OneDrive named **SampleData.xlsx**.
+1. Create a second workbook in OneDrive named **TargetWorkbook.xlsx**.
+1. Open  **SampleData.xlsx**.
+1. Add sample data. You can use the script from the [Write a large dataset in batches](#sample-1-write-a-large-dataset-in-batches) section to generate this data.
+1. Create and save both of the following scripts.
+1. Follow the steps under [Power Automate flow: Read and write data in a loop](#power-automate-flow-read-and-write-data-in-a-loop) to create the flow.
+
+### Sample code: Read part of a workbook
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook, startRow: number, batchSize: number) : string[][] {
+    // This sample only reads the first worksheet in the workbook.
+    const sheet = workbook.getWorksheets()[0];
+
+    // Get the boundaries of the range.
+    // Note that we're assuming usedRange is too big to read or write as a single Range.
+    const usedRange = sheet.getUsedRange();
+    const lastColumnIndex = usedRange.getLastColumn().getColumnIndex();
+    const lastRowindex = usedRange.getLastRow().getRowIndex();
+
+    // If we're starting past the last row, exit the script.
+    if (startRow > lastRowindex) {
+        return [[]];
+    }
+
+    // Get the next batch or the rest of the rows, whichever is smaller.
+    const rowCountToRead = Math.min(batchSize, (lastRowindex - startRow + 1));
+    const rangeToRead = sheet.getRangeByIndexes(startRow, 0, rowCountToRead, lastColumnIndex + 1);
+    return rangeToRead.getValues() as string[][];
+}
+
+```
+
+### Sample code: Write part of a workbook
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook, data: string[][], currentRow: number, batchSize: number): boolean {
+  // Get the first worksheet.
+  const sheet = workbook.getWorksheets()[0];
+
+  // Set the given data.
+  if (data && data.length > 0) {
+    sheet.getRangeByIndexes(currentRow, 0, data.length, data[0].length).setValues(data);
+  }
+
+  // If we wrote less data than the batch size, signal the end of the flow.
+  return batchSize > data.length;
+}
+```
+
+### Power Automate flow: Read and write data in a loop
+
+1. Sign into [Power Automate](https://flow.microsoft.com) and create a new **Instant cloud flow**.
+1. Choose **Manually trigger a flow** and select **Create**.
+1. Add a **New step** to track the current row being read and written. Make a new **Initialize variable** action with the following values.
+    * **Name**: currentRow
+    * **Type**: Integer
+    * **Value**: 0
+
+    :::image type="content" source="../../images/write-large-dataset-1.png" alt-text="The completed 'Initialize variable' step for the 'currentRow'.":::
+1. Add a **New step** to set the number of rows to be read in a single batch. Depending on the number of columns, this made need to be smaller to avoid the data transfer limits. Make a new **Initialize variable** action with the following values.
```

Comment:
Typo fix
```suggestion
1. Add a **New step** to set the number of rows to be read in a single batch. Depending on the number of columns, this may need to be smaller to avoid the data transfer limits. Make a new **Initialize variable** action with the following values.
```